server/chat-plugins/helptickets.js

```diff
@@ -758,6 +848,165 @@ const pages = {
+					resolution: splitLine[4],
+					result: splitLine[5],
+					staff: splitLine[6],
+				};
+			});
+			if (table === 'tickets') {
+				// {total: num, initwait: num, wait: num, resolution: {dead: num, unresolved: num, resolved: num}, result: num positive, totaltickets: num}
+				/** @type {any} */
+				let typeStats = {};
+				for (let stats of ticketStats) {
+					if (!typeStats[stats.type]) typeStats[stats.type] = {total: 0, initwait: 0, wait: 0, resolution: {}, result: 0, totaltickets: 0};
+					let type = typeStats[stats.type];
+					type.totaltickets++;
+					type.total += stats.total;
+					type.initwait += stats.initwait;
+					type.wait += stats.wait;
+					if (['approved', 'valid', 'assisted'].includes(stats.result.toString())) type.result++;
+					if (['dead', 'unresolved', 'resolved'].includes(stats.resolution.toString())) {
+						if (!type.resolution[stats.resolution.toString()]) type.resolution[stats.resolution.toString()] = 0;
+						type.resolution[stats.resolution.toString()]++;
+					}
+				}
+
+				// Calculate averages
+				for (let t in typeStats) {
+					let type = typeStats[t];
+					type.total = Math.round(type.total / type.totaltickets);
+					type.initwait = Math.round(type.initwait / type.totaltickets);
+					type.wait = Math.round(type.wait / type.totaltickets);
+					type.result = Math.round((type.result / type.totaltickets) * 100);
+					for (let resolution in type.resolution) {
+						type.resolution[resolution] = Math.round((type.resolution[resolution] / type.totaltickets) * 100);
+					}
+				}
+
+				let sortedStats = Object.keys(typeStats).sort((a, b) => {
+					if (col === 'type') {
+						// Alphabatize strings
+						if (a < b) return -1;
+						if (a > b) return 1;
+						return 0;
+					} else if (col === 'resolution') {
+						return (typeStats[b][col].resolved || 0) - (typeStats[a][col].resolved || 0);
+					}
+					return typeStats[b][col] - typeStats[a][col];
+				});
+
+				for (let type of sortedStats) {
+					let resolution = Object.keys(typeStats[type].resolution).map(x => {
+						return `${x.charAt(0).toUpperCase() + x.slice(1)}: ${typeStats[type].resolution[x]}%`;
+					}).join(' ');
+					buf += `<tr><td>${type}</td><td>${typeStats[type].totaltickets}</td><td>${Chat.toDurationString(typeStats[type].total, {hhmmss: true})}</td><td>${Chat.toDurationString(typeStats[type].initwait, {hhmmss: true}) || '-'}</td><td>${Chat.toDurationString(typeStats[type].wait, {hhmmss: true}) || '-'}</td><td>${resolution}</td><td>${typeStats[type].result}%</td></tr>`;
+				}
+			} else {
+				/** @type {{[key: string]: {[key: string]: number}}} */
+				let staffStats = {};
+				for (let stats of ticketStats) {
+					let staff = (typeof stats.staff === 'string' ? stats.staff.split(';') : []);
+					for (let s = 0; s < staff.length; s++) {
+						if (!staff[s]) continue;
+						if (!staffStats[staff[s]]) staffStats[staff[s]] = {num: 0, time: 0};
+						staffStats[staff[s]].num++;
+						// @ts-ignore
```

Comment:
what are you ignoring? please either add a comment explaining why necessary or remove the ignore

Reply:
Apparently its ok now? TS was throwing for a weird reason when I coded this but removing the check doesn't cause an error.